### PR TITLE
merge article pagination and pagination, improve spacing

### LINKED ISSUE #3165: Pagination: Previous / Next buttons should support word + chevron for cases when the number of pages is unknown
Context: 
Some search results pages are not able to query the exact number of pages of search results. In those cases, we can't just omit the numbered butons, because the chevron buttons by themselves loose their context and level of prominence. Ideally, we should be able to get the spacing as set out by the icon buttons, by simply placing Previous / Next inside the chevron buttons. This however causes some spacing issues with the following markup:

```
<ol class="p-pagination">
  <li class="p-pagination__item">
    <a class="p-pagination__link--previous" href="#previous" title="Previous page"><i class="p-icon--contextual-menu">Previous page</i>Previous</a>
  </li>
  <li class="p-pagination__item">
    <a class="p-pagination__link--next" href="#next" title="Next page">Next<i class="p-icon--contextual-menu">Next page</i></a>
  </li>
</ol>
```

![image.png]

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

- Ensure pagination chevrons can use text labels without causing spacing issues
- Refactor pagination buttons to use the button placeholders instead of separate styling
- Remove the article pagination page, and replace it with a mention on the pagination component documentation
- Update the doc copy to explain when to use pagination, and when to use article pagination

Fixes #3165 #3167

## QA

- Pull code
- Run `./run`
- Open /docs/patterns/pagination, proofread.
- Open /docs/examples/patterns/pagination/pagination-verbose, paste the contenrt of this [pastebin](https://pastebin.canonical.com/p/BcCdX8q5zd/plain/). Ensure spacing matches the icon button spacing

## Screenshots

Before:

image

After:

image

### CODE REVIEW

**Comment by bartaz** [comment]
The active state (`is-active`) doesn't work anymore.

Before:
Screenshot 2020-07-14 at 14 51 18

After:
Screenshot 2020-07-14 at 14 51 56

**Comment by bartaz** [comment]
There is a double margin bottom now under pagination - both on buttons and on pagination itself:

Screenshot 2020-07-14 at 14 55 26

File: templates/docs/examples/patterns/pagination/pagination-verbose.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "_layouts/examples.html" %} {% block title %}Pagination / Previous and next buttons only {% endblock %} {% block standalone_css %}patterns_pagination{% endblock %} {% block content %}
+<ol class="p-pagination">
+  <li class="p-pagination__item">
+    <a class="p-pagination__link--previous" href="#previous" title="Previous page"><i class="p-icon--contextual-menu">Previous page</i><span>Previous page</span></a>
```
Comment (bartaz) [inline templates/docs/examples/patterns/pagination/pagination-verbose.html:4]: "Previous page" is unnecessarily doubled in the icon.

```suggestion
    <a class="p-pagination__link--previous" href="#previous" title="Previous page"><i class="p-icon--contextual-menu"></i><span>Previous page</span></a>
```

File: templates/docs/examples/patterns/pagination/pagination-verbose.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  </li>
+  <li class="p-pagination__item">
+    <a class="p-pagination__link--next" href="#next" title="Next page"><span>Next page</span><i class="p-icon--contextual-menu">Next page</i></a>
```
Comment (bartaz) [inline templates/docs/examples/patterns/pagination/pagination-verbose.html:7]: Same here:

```suggestion
    <a class="p-pagination__link--next" href="#next" title="Next page"><span>Next page</span><i class="p-icon--contextual-menu"></i></a>
```

Comment (lyubomir-popov) [reply templates/docs/examples/patterns/pagination/pagination-verbose.html]: Fixed

File: templates/docs/examples/patterns/pagination/pagination-verbose.html (hunk 1 above)
Comment (lyubomir-popov) [reply templates/docs/examples/patterns/pagination/pagination-verbose.html]: Fixed

File: templates/docs/examples/patterns/pagination/pagination-verbose.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% extends "_layouts/examples.html" %} {% block title %}Pagination / Previous and next buttons only {% endblock %} {% block standalone_css %}patterns_pagination{% endblock %} {% block content %}
```
Comment (bartaz) [inline templates/docs/examples/patterns/pagination/pagination-verbose.html:1]: It looks like you've run prettier on this.
Could you format it as in other examples (so the tags are in separate lines)?

File: templates/docs/patterns/pagination.md [hunk 4]
```diff
@@ -44,8 +52,21 @@ To import just this component into your project, copy the snippet below and incl
 @include vf-p-icon-contextual-menu;
 ```
 
-For more information see [Customising Vanilla](/docs/customising-vanilla/) in your projects, which includes overrides and importing instructions.
+### Providing context
```
Comment (bartaz) [inline templates/docs/patterns/pagination.md:55]: We do refer to this pattern as `article pagination` in the docs, so would be good if heading uses that, so it shows up in side navigation.

**Pull request merged**